flink-python/pyflink/table/expressions.py

```diff
@@ -0,0 +1,543 @@
+           'UNBOUNDED_RANGE', 'CURRENT_ROW', 'CURRENT_RANGE', 'current_date', 'current_time',
+           'current_timestamp', 'local_time', 'local_timestamp', 'temporal_overlaps',
+           'date_format', 'timestamp_diff', 'array', 'row', 'map_', 'row_interval', 'pi', 'e',
+           'rand', 'rand_integer', 'atan2', 'negative', 'concat', 'concat_ws', 'uuid', 'null_of',
+           'log', 'with_columns', 'without_columns', 'call']
+
+
+def _leaf_op(op_name: str):
+    gateway = get_gateway()
+    return Expression(getattr(gateway.jvm.Expressions, op_name)())
+
+
+def _unary_op(op_name: str, arg):
+    gateway = get_gateway()
+    return Expression(getattr(gateway.jvm.Expressions, op_name)(_get_java_expression(arg)))
+
+
+def _binary_op(op_name: str, first, second):
+    gateway = get_gateway()
+    return Expression(getattr(gateway.jvm.Expressions, op_name)(
+        _get_java_expression(first),
+        _get_java_expression(second)))
+
+
+def _ternary_op(op_name: str, first, second, third):
+    gateway = get_gateway()
+    return Expression(getattr(gateway.jvm.Expressions, op_name)(
+        _get_java_expression(first),
+        _get_java_expression(second),
+        _get_java_expression(third)))
+
+
+def _quaternion_op(op_name: str, first, second, third, forth):
+    gateway = get_gateway()
+    return Expression(getattr(gateway.jvm.Expressions, op_name)(
+        _get_java_expression(first),
+        _get_java_expression(second),
+        _get_java_expression(third),
+        _get_java_expression(forth)))
+
+
+def _add_version_doc():
+    from inspect import getmembers, isfunction
+    from pyflink.table import expressions
+    for o in getmembers(expressions):
+        if isfunction(o[1]) and not o[0].startswith('_'):
+            add_version_doc(o[1], "1.12.0")
+
+
+def col(name: str):
+    """
+    Creates an expression which refers to a table's field.
+
+    Example:
+    ::
+
+        >>> tab.select(col("key"), col("value"))
+
+    :param name: the field name to refer to
+    """
+    return _unary_op("$", name)
+
+
+def lit(v, data_type: DataType = None):
+    """
+    Creates a SQL literal.
+
+    The data type is derived from the object's class and its value. For example, `lit(12)` leads
+    to `INT`, `lit("abc")` leads to `CHAR(3)`.
+
+    Example:
+    ::
+
+        >>> tab.select(col("key"), lit("abc"))
+    """
+    if data_type is None:
+        return _unary_op("lit", v)
+    else:
+        return _binary_op("lit", v, _to_java_data_type(data_type))
+
+
+def range_(start: Union[str, int], end: Union[str, int]):
+    """
+    Indicates a range from 'start' to 'end', which can be used in columns selection.
+
+    Example:
+    ::
+
+        >>> tab.select(with_columns(range_('b', 'c')))
+
+    .. seealso:: :func:`~pyflink.table.expressions.with_columns`
+    """
+    return _binary_op("range", start, end)
+
+
+def and_(predicate0: Union[bool, 'Expression[bool]'],
```

Comment:
Union[bool, 'Expression[bool]'] -> 'Union[bool, Expression[bool]]'?
It would be better if we unify the writing.